sale_import_base/models/sale_import_payload.py

```diff
@@ -0,0 +1,114 @@
+# Copyright 2022 Akretion (https://www.akretion.com).
+# @author Sébastien BEAU <sebastien.beau@akretion.com>
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+
+
+import json
+import traceback
+from datetime import date, timedelta
+
+from psycopg2 import OperationalError
+
+from odoo import api, fields, models
+from odoo.service.model import PG_CONCURRENCY_ERRORS_TO_RETRY
+
+from odoo.addons.queue_job.exception import RetryableJobError
+
+# Use to bypass job entirely for easier debugging
+DEBUG_MODE = False
+
+
+class SaleImportPayload(models.Model):
+    _name = "sale.import.payload"
+    _inherit = ["mail.thread"]
+    _description = "Sale Payload"
+    _order = "id desc"
+    _removal_interval = 365
+
+    data_str = fields.Text(string="Editable data")
+    state = fields.Selection(
+        [("pending", "Pending"), ("done", "Done"), ("fail", "Failed")],
+        default="pending",
+        readonly=True,
+        tracking="1",
+    )
+    state_info = fields.Text(
+        "Additional state information", readonly=True, tracking="1"
+    )
+    sale_channel_id = fields.Many2one(
+        "sale.channel",
+        readonly=True,
+    )
+    company_id = fields.Many2one(
+        "res.company",
+        related="sale_channel_id.company_id",
+        store=True,
+    )
+    stack_trace = fields.Text(readonly=True)
+
+    @api.autovacuum
+    def _delete_old_sale_importer_chunk(self):
+        deletion_limit_date = date.today() - timedelta(days=self._removal_interval)
+        payloads = self.search(
+            [
+                ("create_date", "<", deletion_limit_date),
+                ("state", "=", "done"),
+            ]
+        )
+        payloads.unlink()
+
+    @api.model_create_multi
+    def create(self, vals):
+        # we use mail.thread to log changes in fields but no need of all tracking stuff
+        # on creation for this technical model
+        result = super(
+            SaleImportPayload, self.with_context(tracking_disable=True)
+        ).create(vals)
+        for rec in result:
+            rec.enqueue_job()
+        return result
+
+    def button_retry(self):
+        self.enqueue_job()
+
+    def enqueue_job(self):
+        if DEBUG_MODE:
```

Comment:
I remember here that @kevinkhao suggested to use a `if "pdb" in config.get("dev_mode"):` instead of hardcoded constant :
https://github.com/OCA/sale-channel/pull/20#discussion_r1616647264

What do you think @florian-dacosta ?

Reply:
I think this part is quite ugly, and I am wondering if it should not simply be removed.
But well, let's keep it for now, I replaced it with @kevinkhao suggestion, seems like a good idea to avoid changing the code during debug indeed.